src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterFactory.cs

```diff
@@ -33,6 +33,6 @@ internal JsonConverter GetConverterInternal(Type typeToConvert, JsonSerializerOp
         /// <returns>
         /// An instance of a <see cref="JsonConverter{T}"/> where T is compatible with <paramref name="typeToConvert"/>.
         /// </returns>
-        public abstract JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options);
+        public abstract JsonConverter? CreateConverter(Type typeToConvert, JsonSerializerOptions options);
```

Comment:
`CreateConverter` shouldn't return nullable. It can, if someone implements it incorrectly, which is why we need to guard against it. However, the public API shouldn't state that.

Reply:
It was non null, made it nullable because of your comment https://github.com/dotnet/runtime/pull/528#discussion_r355782202

Reply:
That comment is about the internal `GetConverterInternal` method, not the public abstract `CreateConverter` method. The public API shouldn't return nullable.

Having the internal method being annotated as nullable to help make sure that the serializer does null checks has some value (at least to me, and we can change it non-nullable later)
. 
Making the public `CreateConverter` nullable sends the wrong message about the API contract.

Reply:
Understand and agree that CreateConverter shouldn't return null, just saying there is only one statement calling this method and nothing else. Doesn't make sense for me caller return nullable, called one not. Anyways will update

Reply:
> Doesn't make sense for me caller return nullable, called one not. Anyways will update

Since the caller is an internal method, I am less worried about it (I am OK with the caller to return non-nullable too, if you think that is better). This can be considered the same case as argument validation where we still do a null check even if a parameter is marked non-nullable. The code that calls `GetConverterInternal` just has to make sure to guard against null.